### PR TITLE
Add packets for almost all 2014-2017 clients

### PR DESCRIPTION
**Pull request by 4144**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Add missing shuffle packets for 2014-2017 clients.
Add non shuffle packets for new 2017 clients.
Add encryption keys for all clients starting from 20101123 to 20170920.
Fix char server packets for clients >= 20170830

Latest 2017 clients almost not tested, but look like allow connecting to server.

Packets source in
https://gitlab.com/4144/servergreps/blob/master/client_pre/packets/
https://gitlab.com/4144/servergreps/blob/master/client_pe/packets/

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1844)

**Comment by Asheraf** [comment]
Tested the update for char server packet on `2017-09-13bRagexeRE` and `2017-09-20bRagexeRE` both clients working and exp values show correctly on char select screen.

File: src/map/packets_keys.h [hunk 1]
```diff
@@ -26,496 +26,2110 @@
...
 
+// 2017-05-17aRagexeRE, 2017-05-17bRagexeRE
 #if PACKETVER == 20170517
-	packetKeys(0x2CC4749A,0x1FA954DC,0x72276857); /* 4144 */
+	packetKeys(0x2CC4749A,0x1FA954DC,0x72276857);
+#endif
+
+// 2017-05-24aRagexeRE
+#if PACKETVER == 20170524
+	packetKeys(0x0B4E03A6,0x2
```
Comment (Lemongrass3110) [inline src/map/packets_keys.h]: You lost the custom key definitions here. Intended?

Comment (4144) [reply src/map/packets_keys.h]: hm, no. i will return it back. thanks

**Pull request merged**